src/Diffusion.jl

```diff
@@ -108,4 +112,94 @@ function nmolabsorb(σ, steps, pos, r, molecules) #simulates multiple molecules
     return molhistories
 end
 
+### Analysis ###
+ 
+function molremaining(σ, steps, pos, r, molecules, boundary)
+    if boundary == "absorb"
+        trial = nmolabsorb(σ, steps, pos, r, molecules)
+    elseif boundary == "reflect"
+        trial = nmolreflect(σ, steps, pos, r, molecules)
+    else
+        throw(DomainError("Please specify boudary condition as 'absorb' or 'reflect'"))
+    end
+    lengths = molpathlengths(trial)
+    return percentremaining(lengths)
+    end
```

Comment:
Delete/fix the indentation here and below. While it's not important for the code itself, for humans it makes your code harder to read when it isn't following customary indentation rules. You could try https://github.com/domluna/JuliaFormatter.jl if you don't want to do it manually.